classes/class.attachments.php

```diff
@@ -1003,27 +1003,19 @@ function get_field_types() {
             foreach ( $field_types as $type => $path ) {
                 // proceed with inclusion
                 if ( file_exists( $path ) ) {
+                    // Store a list of all declared classes before we include the file
+                    $existing_classes = get_declared_classes();
+                    
                     // include the file
                     include_once( $path );
 
-                    // store the registered classes so we can single out what gets added
-                    $existing_classes = get_declared_classes();
-
-                    // we're going to use our Attachments class as a reference because
-                    // during subsequent instantiations of Attachments (e.g. within template files)
-                    // these field classes WILL NOT be added to the array again because
-                    // we're using include_once() so that strategy is no longer useful
```

Comment:
Wouldn't your method fail in exactly the described case (file was included before)?